src/dependencies/index.ts

```diff
@@ -20,11 +18,18 @@ export async function installDependencies(context: vscode.ExtensionContext, shou
 
         // TODO: Stop prusti-rustc and cargo-prusti
 
-        const tools = prustiTools(currentPlatform!, context);
-        const { result: location, didReportProgress } = await withProgressInWindow(
+        const deps = prustiTools(tools.currentPlatform!, context);
+        const { result, didReportProgress } = await tools.withProgressInWindow(
             `${shouldUpdate ? "Updating" : "Installing"} Prusti`,
-            listener => tools.install(config.buildChannel(), shouldUpdate, listener)
+            listener => deps.install(config.buildChannel(), shouldUpdate, listener)
         );
+        if (!(result instanceof tools.Success)) {
+            util.userError(
+                "Prusti installation has been canceled. Please restart the IDE to retry.",
+                true, verificationStatus
+            )
+        }
+        const location = (result as tools.Success<tools.Location>).value;
```

Comment:
@ArquintL Do you know a better way to pattern match the `InstallResult<Location>`? My eslint is not happy with this line.

Reply:
Based on line 26, my typechecker (and hopefully eslint too) infers that `result` is of type `tools.Success<any>`. Have you tried to use `result.value as tools.Location`?

Reply:
Thanks! The branch at line `26` was missing a return. After adding that, `result.value as tools.Location` works perfectly.

Reply:
Great, I've missed the missing return too